sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -3437,21 +3468,38 @@ pi_result cuda_piextUSMSharedAlloc(void **result_ptr, pi_context context,
                                    pi_device device,
                                    pi_usm_mem_properties *properties,
                                    size_t size, pi_uint32 alignment) {
+  static constexpr pi_uint32 max_alignment =
+      0x200; // from empirical testing with CUDA 10.2 on a Tesla K40
+
+  // enforce a valid pointer to the allocated memory
   assert(result_ptr != nullptr);
+  // check the the context is valid
   assert(context != nullptr);
+  // check that the device is valid
   assert(device != nullptr);
+  // check that the property list is empty
   assert(properties == nullptr);
+  // check that the alignment is not larger than max_alignment, and is either 0
+  // or a power of 2
+  assert(alignment <= max_alignment && __builtin_popcount(alignment) <= 1);
+
   pi_result result = PI_SUCCESS;
   try {
     ScopedContext active(context);
-    result = PI_CHECK_ERROR(cuMemAllocManaged((CUdeviceptr *)result_ptr, size,
-                                              CU_MEM_ATTACH_GLOBAL));
-  } catch (pi_result error) {
-    result = error;
+
+    pi_result result = PI_SUCCESS;
+    try {
+      ScopedContext active(context);
+      result = PI_CHECK_ERROR(cuMemAllocManaged((CUdeviceptr *)result_ptr, size,
+                                                CU_MEM_ATTACH_GLOBAL));
+    } catch (pi_result error) {
+      result = error;
+    }
+    // check that the result is suitable aligned
+    assert((alignment == 0) ||
+           (reinterpret_cast<std::uintptr_t>(*result_ptr) % alignment == 0));
+    return result;
   }
-  assert(reinterpret_cast<std::uintptr_t>(*result_ptr) % alignment == 0);
-  return result;
-}
```

Comment:
Missing `}` is breaking the build.

Reply:
Actually there were 3 stray copy-pasted lines, sorry about that :-(